element-connector/src/main/java/org/eclipse/californium/elements/RelaxedCorrelationContextMatcher.java

```diff
@@ -36,29 +36,19 @@ public boolean isResponseRelatedToRequest(CorrelationContext requestContext, Cor
 		return internalMatch(requestContext, responseContext);
 	}
 
+	@Override
+	public boolean isToBeSent(CorrelationContext messageContext, CorrelationContext connectorContext) {
+		return internalMatch(messageContext, connectorContext);
+	}
+
 	private final boolean internalMatch(CorrelationContext requestedContext, CorrelationContext availableContext) {
 		if (null == requestedContext) {
 			return true;
 		} else if (null == availableContext) {
 			return false;
 		}
 		if (requestedContext.get(DtlsCorrelationContext.KEY_SESSION_ID) != null) {
-			boolean match = requestedContext.get(DtlsCorrelationContext.KEY_SESSION_ID).equals(
-					availableContext.get(DtlsCorrelationContext.KEY_SESSION_ID))
-					&& requestedContext.get(DtlsCorrelationContext.KEY_CIPHER).equals(
-							availableContext.get(DtlsCorrelationContext.KEY_CIPHER));
-
-			LOGGER.log(
-					match ? Level.FINEST : Level.WARNING,
-					"(D)TLS session {0}, {1}",
-					new Object[] { requestedContext.get(DtlsCorrelationContext.KEY_SESSION_ID),
-							availableContext.get(DtlsCorrelationContext.KEY_SESSION_ID) });
-			LOGGER.log(
-					match ? Level.FINEST : Level.WARNING,
-					"(D)TLS cipher {0}, {1}",
-					new Object[] { requestedContext.get(DtlsCorrelationContext.KEY_CIPHER),
-							availableContext.get(DtlsCorrelationContext.KEY_CIPHER) });
-			return match;
+			return CorrelationContextUtil.match(getName(), KEYS, requestedContext, availableContext);
```

Comment:
I guess the points are simply, that
- I would like to consider the TCP stuff (TCP_CONNECTION_ID) and, 
- when "send" would be extended to provide a correlation context, I would like to grant, that, if someone "outside" make a misstake and send data using a DTLS correlation context, but gets a plain connector, the message is dropped. 

I agree, that "californium internal", responses are sent via the connector the request comes in. and TCP is not in the focus. But I hope, the "send" extension will come soon :-).